src/build/env.py

```diff
@@ -51,191 +24,166 @@ class IsolatedEnvironment(object):
     Non-standard paths injected directly to sys.path still be passed to the environment.
     """
 
-    _MANIPULATE_PATHS = ('purelib', 'platlib')
+    def __init__(self, path, executable, install_executable):
+        # type: (str, str, str) -> None
+        """
+        Define an isolated build environment.
 
-    def __init__(self, remove_paths, _executable=sys.executable):  # type: (Sequence[str], str) -> None
+        :param path: the path where the environment exists
+        :param executable: the python executable within the environment
+        :param install_executable: an executable that allows installing packages within the environment
         """
-        :param remove_paths: Import paths that should be removed from the environment
+        self._path = path
+        self._install_executable = install_executable
+        self._executable = executable
+
+    @classmethod
+    def for_current(cls):  # type: () -> IsolatedEnvironment
+        """
+        Create an isolated build environment into a temporary folder that matches the current interpreter.
+
+        :return: the isolated build environment
         """
-        self._env = {}  # type: Dict[str, Optional[str]]
-        self._env_vars = {}  # type: Dict[str, str]
-        self._path = None  # type: Optional[str]
-        self._remove_paths = []  # type: List[str]
-        self._executable = _executable
-        self._old_executable = None  # type: Optional[str]
-        self._pip_executable = None  # type: Optional[str]
-
-        # normalize paths so that we can compare them -- required on case insensitive systems
-        for path in remove_paths:
-            self._remove_paths.append(os.path.normcase(path))
+        path = tempfile.mkdtemp(prefix='build-env-')
+        executable, pip_executable = _create_isolated_env(path)
+        return cls(path=path, executable=executable, install_executable=pip_executable)
+
+    def __enter__(self):  # type: () -> IsolatedEnvironment
+        """Enable the isolated build environment"""
+        return self
```

Comment:
We should adjust `PATH` here so that you can call scripts from the environment.

Reply:
I'm strongly against that, but can you expand on why? 

My reasoning is IMHO mangling with the path is always a really bad idea. See https://bugs.python.org/issue42041 for a recent bug and discussion on why.

Reply:
Sure. Then we should implement a `get_script` method. I can do this in a later PR.

Reply:
Why and where would you need it though? Can you expand on that?

Reply:
I need it if I want to call a script that is provided by a package.

One could argue that you should be able to call entrypoints directly without need for a script, but they are not always public API. And regardless, per PEP 427, packages may bring their own scripts, so I have no other alternative.

Reply:
> I need it if I want to call a script that is provided by a package.

> And regardless, per PEP 427, packages may bring their own scripts, so I have no other alternative than to call the script directly.

The above still doesn't answer the question of where and why would you want to call a script?